domain/application/state/application.go

```diff
@@ -700,6 +748,8 @@ func (st *ApplicationState) SetApplicationLife(ctx domain.AtomicContext, appID c
 UPDATE application
 SET life_id = $applicationID.life_id
 WHERE uuid = $applicationID.uuid
+-- we ensure the life can never go backwards.
+AND life_id <= $applicationID.life_id
```

Comment:
Rather than peacefully ignoring the invalid life change, should we raise an error instead?

Reply:
No, this is to cover the lack of txn isolation. We don'twant to put the burden on the caller since it's a db limitation.